app/scripts/browser.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+var _ = require('underscore');
+
+// Utilities for cross-browser compatibility.
+var Browser = {};
+
+// Map of CSS properties to property names used by the DOM in this browser.
+var normalizedProperties = {
+  'transform': _.find(['transform', 'WebkitTransform'], function (property) {
+    return document.documentElement.style[property] !== undefined;
+  }) || 'transform'
+};
+
+// Convert CSS properties and values to equivalent properties and values for
+// this browser.
+Browser.normalizeStyles = function (styles) {
+  var normalized = {};
+  _.forEach(_.keys(styles), function (property) {
+    if (Object.prototype.hasOwnProperty.call(normalizedProperties, property)) {
```

Comment:
It's less safe to assume that an object being used like a "map" will have `Object.prototype` methods. For instance, `styles` could look like `{hasOwnProperty: 1}`, and this code would break.

Obviously, there is no CSS property called "hasOwnProperty," but when using objects like "maps," I prefer to use them in the context of the map data structure, rather than the particular use case of the data structure.

My definition of a "map" is "a data structure with keys mapping to values." There should be no restriction on which keys can be used in a map. But by using "hasOwnProperty" as a method, implicitly the set of available keys is reduced the set of keys excluding the names of `Object.prototype` methods, which is also an unbounded set, since it's defined by ECMA and could change from one year to the next.

For these reasons, I recommend adopting this safer style of checking for own properties.